basil.js

```diff
@@ -5310,14 +5310,29 @@ var initDataFile = function(file) {
     error(getParentFunctionName(1) + "(), invalid first argument. Use File or a String describing a file path.");
   }
 
+  var fileString = "" + file;
+
   var result = null;
   if (file instanceof File) {
     result = file;
   } else {
-    result = new File(pub.projectFolder().absoluteURI + "/data/" + file);
+
+    // if the document has been saved, check for the file in the data folder
+    if(currentDoc().saved) {
+      result = new File(pub.projectFolder().absoluteURI + "/data/" + file);
+      if(!result.exists) {
+        result = null;
```

Comment:
I guess a good UX would be:

- assume that there is a data folder and look for the file
- if file does not exist look for a path relative to the doc without data folder
- if file exists use it and give info that it should be in data folder
- if file does not exist throw error

Reply:
The data folder thing comes from Processing.
And in p5.js and Processing it works exactly like that: `loadStrings()` allows for either a path inside the data/sketch folder or an absolute path. So that's what we now have as well and I think we should stick to that.

Reply:
I mean, the main difference is that p5.js has the sketch folder itself as the start folder and Processing the data folder. I am getting more and more annoyed at p5.js for not just staying in sync with Processing with many methods.

But since we always worked with the data folder in basil as well and that is how it is implemented in quite a few functions right now, I would also keep the data folder as starting point instead of using the project folder.